be/src/exec/vectorized/json_scanner.cpp

```diff
@@ -707,16 +707,23 @@ Status JsonReader::_read_and_parse_json() {
     StreamLoadPipeInputStream* stream_file = down_cast<StreamLoadPipeInputStream*>(_file->stream().get());
     {
         SCOPED_RAW_TIMER(&_counter->file_read_ns);
-        // For efficiency reasons, simdjson requires a string with a few bytes (simdjson::SIMDJSON_PADDING) at the end.
-        RETURN_IF_ERROR(stream_file->pipe()->read_one_message(&_parser_buf, &_parser_buf_cap, &_parser_buf_sz,
-                                                              simdjson::SIMDJSON_PADDING));
-        if (_parser_buf_sz == 0) {
-            return Status::EndOfFile("EOF of reading file");
+        auto st = stream_file->pipe()->read();
+        if (!st.ok()) {
+            return st.status();
+        }
+        _parser_buf = st.value();
+
+        if (_parser_buf->capacity < _parser_buf->remaining() + simdjson::SIMDJSON_PADDING) {
```

Comment:
is it possible to make sure this when buffer alloc

Reply:
Yes. But if it is done by the producer of `stream_load_pipe`, since there're many producers, it would make the code fragmented. It's not easy to remind every programmer to remember this. Besides, the `simdjson::SIMDJSON_PADDING` only cost several bytes. There's usually extra space in the `ByteBuffer`, so the possibility of re-allocation is low.